codalab/worker/local_run/local_dependency_manager.py

```diff
@@ -87,45 +89,83 @@ def _save_state(self):
             self._state_committer.commit({'dependencies': self._dependencies, 'paths': self._paths})
 
     def _load_state(self):
+        """
+        Load states from dependencies-state.json file and sync them with local directories specified by self.dependencies_dir.
+        Populate values for self._dependencies, self._dependency_locks and self._paths by
+        1. Load states from dependencies-state.json file and local directory self.dependencies_dir.
```

Comment:
'states' is used in a vague way here...be more precise (but succinct) about what is loaded.  In general, this description seems like it's overly complicated.  I think it'd be better to describe the purpose of this function rather than a step by step procedure - what invariants are maintained?  I'd write something like:

There are two pieces of information:
(1) dependencies-state.json: contains bundles with some information (e.g., size, last_used)
(2) local disk: the bundle contents
Ideally, these would be in sync, but life happens and we want to robust to them not being in sync.  Therefore, when we load the state, we keep only the bundles that appear in both (1) and (2).